pkg/cmd/experimental/syncgroups/grouplister.go

```diff
@@ -0,0 +1,103 @@
+package syncgroups
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+	syncutil "github.com/openshift/origin/pkg/cmd/experimental/syncgroups/util"
+)
+
+// NewAllOpenShiftGroupLister returns a new AllLocalGroupLister
+func NewAllOpenShiftGroupLister(host string, groupClient osclient.GroupInterface) interfaces.LDAPGroupLister {
+	return &AllLocalGroupLister{
+		client: groupClient,
+		host:   host,
+	}
+}
+
+// AllLocalGroupLister lists unique identifiers for LDAP lookup of all local OpenShift Groups that
+// have been marked with an LDAP URL annotation as a result of a previous sync.
+type AllLocalGroupLister struct {
+	client osclient.GroupInterface
+	host   string
+}
+
+func (l *AllLocalGroupLister) ListGroups() (ldapGroupUIDs []string, err error) {
+	operator := labels.EqualsOperator
+	val := util.NewStringSet(l.host)
+	requirement, err := labels.NewRequirement(syncutil.LDAPURL, operator, val)
+	if err != nil {
+		return nil, fmt.Errorf("unexpected error gemerating label selector: %v", err)
+	}
+	label := labels.LabelSelector{*requirement}
+	field := fields.Everything()
+
+	potentialGroups, err := l.client.List(label, field)
+	if err != nil {
+		return nil, err
+	}
+
+	groupLister := LocalGroupLister{
+		Groups: potentialGroups.Items,
+	}
+
+	uids, err := groupLister.ListGroups()
+	if err != nil {
+		return nil, err
+	}
+	return uids, nil
+}
+
+// NewWhitelistOpenShiftGroupLister returns a new LocalGroupLister
+func NewWhitelistOpenShiftGroupLister(whitelist []string, client osclient.GroupInterface) interfaces.LDAPGroupLister {
```

Comment:
`openshiftWhitelist` and indicate in godoc that this whitelist is for the openshift groups, not the resulting ldap groupUID